plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/edit/parts/DataMapperRootEditPart.java

```diff
@@ -16,65 +16,285 @@
+            }
+            if ( attributePath.indexOf(nodeName) !=-1) {
+                treeNodeFinal = traverseNode(treeNodeFinal, treeNode, attributeName, attributePath);
+            }
+        }
+        return treeNodeFinal;
+    }
+    
+    private TreeNode traverseNode(TreeNode treeNodeFinal, TreeNode prevTreeNode, String attributeName, String attributePath) {
+        EList<TreeNode> treeNode = prevTreeNode.getNode();
+        Iterator iterator = treeNode.iterator();
+        treeNodeFinal = iterateList(treeNodeFinal, iterator,  attributeName, attributePath);
+        return treeNodeFinal;
+    }
+    
+    private static OutNode deduceContainer(EObject source, EObject target) {
+        for (EObject element = source; element != null; element = element.eContainer()) {
+            if (element instanceof OutNode) {
+                return (OutNode) element;
+            }
+        }
+        return null;
+    }
+    
+    private String getSchema(DataMapperRootImpl datamapperRoot, String type) {
+        ISchemaTransformer schemaTransformer = null;
+        try {
+            schemaTransformer = SchemaTransformerRegistry.getInstance().getSchemaTransformer().newInstance();
+        } catch (InstantiationException | IllegalAccessException e) {
+          log.error(e);
+        }
+        // Model root of input schema tree
+        EList<TreeNode> TreeNodesList = null;
+        if (type.equals("input")){
+            TreeNodesList = ((DataMapperRoot) datamapperRoot).getInput().getTreeNode();
+        }
+        else {
+            TreeNodesList = ((DataMapperRoot) datamapperRoot).getOutput().getTreeNode();
+        }
+        File schemaFile = null;
+        String content = null;
+        // If a tree node is found, continue saving
+        if (null != TreeNodesList && !TreeNodesList.isEmpty()) {
+            TreeNodeImpl inputTreeNode = (TreeNodeImpl) TreeNodesList.get(0);
+            content = schemaTransformer.getSchemaContentFromModel(inputTreeNode, schemaFile, true, content);  //.getSchemaContentFromModel(inputTreeNode, schemaFile);
+        }
+        // Empty tree node, clear the file
+        else {
+            content = "";
+        }
+        return content;
+    }
+    
+    public List<String> getMappings(String inputSchema, String outputSchema) {
+        HttpURLConnection connection = null;  
+        
+        // Create one set of schemas
+        JSONArray schemas = new JSONArray();
+        schemas.put(inputSchema);
+        schemas.put(outputSchema);
+        
+        String schemasToSend = schemas.toString();
+        
+        final HttpClient httpClient = new HttpClient();
+        
+        // Send file
+        StringRequestEntity requestEntity;
+        String outString;
+        List<String> myList = null;
+        try {
+            requestEntity = new StringRequestEntity(schemasToSend,"application/json","UTF-8");
+            PostMethod postMethod = new PostMethod("http://127.0.0.1:5000/uploader");
```

Comment:
We should avoid hard coding the URI at the code level as it is pointing to the local machine. Have we hosted this backend service API somewhere?